### PR TITLE
Add query library behind dev-mode-only feature flag

### PR DESCRIPTION
**Pull request by ArturWierzbicki**
`highq` branch revival:
- scrapped the whole alerting part
- replaced the whole backend part with a simple file-based storage in `data` folder so we can iterate on the UI/UX while waiting for the GA of the Entity API
- `queryLibrary` dev-mode-only flag
- usages tab currently shows all dashboards

### CODE REVIEW

File: public/app/features/query/components/QueryGroup.tsx [hunk 1]
```diff
@@ -220,6 +310,18 @@ export class QueryGroup extends PureComponent<Props, State> {
...
+              <InlineFormLabel htmlFor="saved-query-picker" width={'auto'}>
+                Saved query
+              </InlineFormLabel>
+              <div className={styles.dataSourceRowItem}>
+                <SavedQueryPicker current={this.state.savedQueryUid} onChange={this.onChangeSavedQuery} />
+              </div>
+            </div>
+          </>
+        )}
       </div>
```
Comment (nmarrs) [inline public/app/features/query/components/QueryGroup.tsx:325]: We should carefully review these changes / verify functionality given how core this component is in Grafana

File: public/app/features/query-library/components/SaveQueryWorkflowModal.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+              >
+                {options.workflow === WorkflowID.PR ? 'Submit PR' : 'Push'}
+              </Button>
+            </VerticalGroup>
+          </VerticalGroup>
+        )}
+      </Form>
+    </Modal>
+  );
+};
```
Comment (nmarrs) [inline public/app/features/query-library/components/SaveQueryWorkflowModal.tsx:79]: This component is not currently being used anywhere

File: public/app/features/query-library/components/UsagesTab.tsx [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+      location: item.location,
+      panel_type: item.panel_type,
+    }));
+  }, [searchQuery]);
+
+  if (results.loading) {
+    return <Spinner />;
+  }
+
+  const found = results.value;
```
Comment (nmarrs) [inline public/app/features/query-library/components/UsagesTab.tsx:50]: This feature doesn't seem to be working as expected for me locally

File: public/app/features/dashboard/state/PanelModel.ts [hunk 4]
```diff
@@ -155,6 +171,7 @@ export class PanelModel implements DataConfigSource, IPanelModel {
   datasource: DataSourceRef | null = null;
   thresholds?: any;
   pluginVersion?: string;
+  savedQueryLink: SavedQueryLink | null = null;
```
Comment (ryantxu) [inline public/app/features/dashboard/state/PanelModel.ts:174]: This is the line that makes me a bit nervous -- everything else is well compartmentalized into a more clearly owned/experimental structure.   PanelModel is the core of everything and getting a bunch of attention from the dashboard squad now.

Perhaps the two types defined here can be under: `app/features/query-library/...`

**Review by ryantxu — APPROVED** [review]
looks good -- i added comments to the two files that are in general use

**Review by nmarrs — APPROVED** [review]

**Pull request merged**